public/app/features/explore/utils/links.ts

```diff
@@ -127,10 +129,42 @@ export const getFieldLinksForExplore = (options: {
         }
         return linkModel;
       } else {
+        let internalLinkSpecificVars: ScopedVars = {};
+        if (link.internal?.transformations) {
```

Comment:
I'll create a separate task where we can define some clean-up subtasks, for example, we can move transformations to a separate util/file. We can do it after we add more features to actually see where different parts fit.

Reply:
I split out transformations to its own file because it wasn't that difficult to do and makes sense logically, let me know if this isn't what youre thinking.